framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -836,43 +882,72 @@ void OpenXrReplayConsumerBase::Process_xrPollEvent(const ApiCallInfo&
+                              GetEventTypeString(capture_event->type),
+                              capture_event->type,
+                              capture_event->type);
+
+            // If the event returned is not the one we expect, we may still want to keep track of
+            // it or bail.
             if (capture_event->type != out_eventData->type)
             {
                 if (replay_result == XR_SUCCESS)
                 {
-                    // Add it to the list of received events
-                    received_events_.push_back(*out_eventData);
+                    // We're not yet expecting this event, but we may expect it in the future,
+                    // so keep track of it in the list of previously received events that we
+                    // can check again in the future.
+                    unhandled_events_from_replay_.push_back(*out_eventData);
                     GFXRECON_LOG_WARNING("Recording event for later %s (0x%x, %u)",
-                                         GetEventTypeString(capture_event->type),
-                                         capture_event->type,
-                                         capture_event->type);
-
-                    // If we grow too lare on the event vector, it's probably because we have
-                    // received a bunch of events we can not handle.  So remove the first 100
-                    // events to make room for more without bloating the list size.
-                    // TODO: Perhaps do this more elegantly?
-                    if (received_events_.size() > 1000)
+                                         GetEventTypeString(out_eventData->type),
+                                         out_eventData->type,
+                                         out_eventData->type);
+                    if (ManageEventDequeSize(unhandled_events_from_replay_))
                     {
-                        GFXRECON_LOG_WARNING("Event list is now %d in size, stripping the first 100!",
-                                             received_events_.size());
-                        received_events_.erase(received_events_.begin(), received_events_.begin() + 100);
+                        GFXRECON_LOG_WARNING("Previously received replay event list is size adjusted down to %d",
+                                             unhandled_events_from_replay_.size());
                     }
                 }
                 else if (replay_result == XR_EVENT_UNAVAILABLE)
                 {
-                    // Yield and retry
-                    std::this_thread::sleep_for(std::chrono::nanoseconds(sleep_time));
-
-                    if (sleep_time < kMaxSleepLimitNs)
+                    // If the event that is expected is not known to  us, and we didn't encounter it , then this is a
+                    // bad situtation and needs to be reported.
+                    if (IsHandledEventType(capture_event->type))
```

Comment:
Why don't we retry for unknown events, especially if we think them, likely and valid?

Reply:
Originally, I did, but I thought you implied in a previous review that I should not be.  I'll change it back.